proposals/lip-0050.md

```diff
@@ -171,14 +239,47 @@ registerBLSKeyParamsSchema = {
         "proofOfPossession": {
             "dataType": "bytes",
             "fieldNumber": 2
+        },
+        "generatorKey": {
+            "dataType": "bytes",
+            "fieldNumber": 3
         }
     }
 }
 ```
 
+#### Verification
+
+```python
+def verify(trs: Transaction) -> None:
+    validatorAddress = 20-byte address derived from trs.senderPublicKey
+
+    if validator.getValidatorAccount(validatorAddress) does not exists :
+        raise Exception('Public key does not correspond to a registered validator.')
+        
+    validatorAccount = validator.getValidatorAccount(validatorAddress)
+    if (validatorAccount.generatorKey != INVALID_ED25519_KEY
+        and validatorAccount.generatorKey == trs.params.generatorKey
```

Comment:
Could it be somehow that the validator account doesn't have `generatorKey` set? Also, I don't understand this line of the `if` condition, if `validatorAccount.generatorKey == trs.params.generatorKey` then input generator key equals the one set in the store.
I think of a condition like this: ```if validatorAccount.generatorKey exists and validatorAccount.generatorKey != INVALID_ED25519_KEY```.

Reply:
Maybe it should be `!=` here assuming that if your generator key is already set, then you have to put your existing generator key in the transaction.

Reply:
The generator key always exists (as do other properties of our schemas), the question is only:
- does it equal the INVALID key set during migration?
if not, the transaction sender should provide the already set key.

As Jan mentioned, we could relax this requirement and always allow this transaction to set the generator key. Even in the case when it is already set.

Reply:
Updated to always allow to update the generator key, if the BLS key does not already exist.

Reply:
Can be resolved from my side.